### PR TITLE
Free room

### PR DESCRIPTION
**Pull request by quirk0o**
@adamzima @pbzymek @s1mplex @jtkpiotr @lkudyba @mhib

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -30,7 +30,9 @@ class EventsController < ApplicationController
   end
 
   def index
-    render json: GoogleEventLister.new(session[:credentials], session[:email]).call(TimeInterval.week(date_param))
+    events = GoogleEventLister.new(session[:credentials], session[:email]).call(span_param)
+    return render json: events unless params[:grouped]
```
Comment (pbzymek) [inline app/controllers/events_controller.rb]: render json: !params[:grouped] ? events : EventGrouper.new(events).build_blocks

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -55,4 +57,9 @@ def event_params
   def date_param
     params[:date] ? Date.parse(params[:date]) : Date.today
```
Comment (pbzymek) [inline app/controllers/events_controller.rb]:
``` ruby
def date_param
  Date.parse(params[:date])
rescue 
  Date.today
end
```

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -55,4 +57,9 @@ def event_params
...
   end
+
+  def span_param
+    return TimeInterval.week(date_param) unless params[:start] && params[:end]
```
Comment (pbzymek) [inline app/controllers/events_controller.rb]:
``` ruby
def span_param
  TimeInterval.new(Time.parse(params[:start]), Time.parse(params[:end]))
rescue 
  TimeInterval.week(date_param)
end
```

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (adamzima) [inline app/controllers/events_controller.rb]:
```
render json: params[:grouped] ? EventGrouper.new(events).build_blocks : events
```

:P

Btw, why do we still return grouped events?

File: client/app/components/modal/CreateEventModal.jsx [hunk 4]
```diff
@@ -82,36 +99,85 @@ export default class CreateEventModal extends React.Component {
...
+  _disableSaving() {
+    return keys(this.state.errors).length > 0;
+  }
+
+  _updateRoomAvailability() {
+    const { startTime, endTime } = this.state;
+    EventSource
+      .fetch({ start: startTime, end: endTime })
+      .then(({ data }) => {
+        const unavailableRooms = flow(
```
Comment (adamzima) [inline client/app/components/modal/CreateEventModal.jsx:172]: nice :+1:

File: client/app/components/modal/layout/ModalBody.jsx [hunk 5]
```diff
@@ -1,18 +1,20 @@
...
 
 export default class ModalBody extends React.Component {
   static propTypes = {
-    updateParam:      func.isRequired,
-    conferenceRooms:  array.isRequired,
-    errors:           object,
+    updateParam: func.isRequired,
+    availableLocations: array.isRequired,
+    unavailableLocations: array.isRequired,
+    selectedLocation: number,
```
Comment (adamzima) [inline client/app/components/modal/layout/ModalBody.jsx:16]: Rails enum?

Comment (adamzima) [inline client/app/components/modal/layout/ModalBody.jsx:16]: yep

Comment (adamzima) [inline client/app/components/modal/layout/ModalBody.jsx:16]: thanks

Comment (adamzima) [inline client/app/components/modal/layout/ModalBody.jsx:16]: May consider having separate PropType for this one since it's an enum.

**Pull request merged**